articles/agora-cloud-recording/index.md

```diff
@@ -0,0 +1,496 @@
+### Cloud recording vs. On-Premise recording.
+[Cloud recording](https://docs.agora.io/en/cloud-recording/landing-page?platform=RESTful) is used to record and save voice calls, video calls, and interactive streaming on your cloud storage. You can record one-to-one or one-to-many audio and video calls. For cloud recording, Agora provides APIs to record the streams. Once the recording ends, Agora will upload the recorded video to your cloud storage.
+
+Agora supports uploads to Amazon S3, Qiniu Cloud, Alibaba Cloud, Tencent Cloud, Kingsoft Cloud.
+
+I recommend using [Amazon S3](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-bucket.html) as it is easy to set up. Agora will need the bucket name, the access key, and the secret key for the bucket to upload the recorded files to your cloud storage.
+
+[On-Premise recording](https://docs.agora.io/en/Recording/product_recording?platform=Linux) is similar to cloud recording, but you need to set up your own Linux server using the components provided by Agora to record the streams.
+
+Compared with On-premise Recording, Cloud Recording is more convenient as it does not require deploying Linux servers.
+
+If you'd like to learn more about the differences between cloud recording and on-premise recording, refer to this [FAQ answer](https://docs.agora.io/en/All/faq/onpremise_cloud).
+
+Agora will bill you for the cloud recording service. You can check their [billing policies](https://docs.agora.io/en/cloud-recording/billing_cloud_recording?platform=RESTful) for more information.
+
+### Enabling cloud recording in the project management console
+To use cloud recording in your application, you must enable cloud recording from the Agora Project Management Console.
+
+Head to the console and open Project Management.
+
+Click on usage for the project you want to enable cloud recording for.
+
+![Usage Button](usage_button.png)
+
+Click on duration under cloud recording. If you haven't already enabled it yet, It will ask you whether you want to enable cloud recording for this project. Click on the enable button.
+
+![Enable Cloud Recording](enable_cr.png)
+
+It'll display the maximum number of channels that you can record at the same time. The default value is 50 channels. If you want more, you need to contact the Sales team of Agora (sales@agora.io).
+
+![Max Concurrent Channels](max_concurrent_channels.png)
+
+Once you hit Apply, you'll see the statistics page of the cloud recording.
+
+![Cloud Recording Enabled](cr_enabled.png)
+
+You have successfully enabled cloud recording for your project.
+
+### Acquiring authentication keys for Agora APIs
+If you want to work with Agora's APIs, you need to acquire the client ID and client secret from Agora.
+
+Head to the console and click the account name on the top right corner and click on RESTful API from the dropdown.
+
+![RESTful API Dropdown](restful_api.png)
+
+Copy the customer ID and click on download under customer secret to get the customer secret.
```

Comment:
I notice that this CUSTOMER ID we get is not used in the entire app
But the APP ID which is used severally is not obtained at any one point in the tutorial

Reply:
The Customer ID is only used in generating the Base64 Auth token. The App ID, however, is used in all the requests.

The reason I didn't include how to obtain the APP ID is that if someone is actually going through this tutorial to implement it, I assumed they must have an Agora application already which definitely requires an App ID and they must already have it.

I'll add how to acquire an App ID now.

Reply:
I added a step on how to acquire the App ID